From 1341292cc42ba9481cc47a5456d5fc0646d3bad0 Mon Sep 17 00:00:00 2001
From: Artem Polishchuk <ego.cordatus@gmail.com>
Date: Sun, 12 Dec 2021 19:44:04 +0300
Subject: [PATCH 1/3] Add AppStream metainfo

Infrastructure for distro-agnostic software-centers and universal software component metadata.

https://www.freedesktop.org/software/appstream/docs/
---
 qt/org.openrgb.OpenRGB.metainfo.xml | 77 +++++++++++++++++++++++++++++
 1 file changed, 77 insertions(+)
 create mode 100644 qt/org.openrgb.OpenRGB.metainfo.xml

diff --git a/qt/org.openrgb.OpenRGB.metainfo.xml b/qt/org.openrgb.OpenRGB.metainfo.xml
new file mode 100644
index 00000000..c49a93ba
--- /dev/null
+++ b/qt/org.openrgb.OpenRGB.metainfo.xml
@@ -0,0 +1,77 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2021 Artem Polishchuk <ego.cordatus@gmail.com> -->
+<component type="desktop-application">
+  <id>org.openrgb.OpenRGB</id>
+  <launchable type="desktop-id">OpenRGB.desktop</launchable>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>GPL-2.0-or-later</project_license>
+
+  <name>OpenRGB</name>
+  <provides>
+    <binary>openrgb</binary>
+  </provides>
+
+  <content_rating type="oars-1.0">
+  </content_rating>
+
+  <summary>
+    Open source RGB lighting control that doesn't depend on manufacturer
+    software.
+  </summary>
+
+  <description>
+    <p>
+      One of the biggest complaints about RGB is the software ecosystem
+      surrounding it.  Every manufacturer has their own app, their own brand,
+      their own style.  If you want to mix and match devices, you end up with a
+      ton of conflicting, functionally identical apps competing for your
+      background resources.  On top of that, these apps are proprietary and
+      Windows-only.  Some even require online accounts.  What if there was a way
+      to control all of your RGB devices from a single app, on both Windows and
+      Linux, without any nonsense?  That is what OpenRGB sets out to achieve.
+      One app to rule them all.
+    </p><p>
+      Features
+    </p>
+    <ul>
+      <li>Set colors and select effect modes for a wide variety of RGB
+      hardware</li>
+      <li>Save and load profiles</li>
+      <li>Control lighting from third party software using the OpenRGB SDK</li>
+      <li>Command line interface</li>
+      <li>Connect multiple instances of OpenRGB to synchronize lighting
+      across multiple PCs</li>
+      <li>Can operate standalone or in a client/headless server configuration</li>
+      <li>View device information</li>
+      <li>No official/manufacturer software required</li>
+      <li>Graphical view of device LEDs makes creating custom patterns easy</li>
+    </ul>
+  </description>
+
+  <releases>
+    <release version="0.6" date="2021-06-01"></release>
+    <release version="0.5" date="2021-12-07"></release>
+    <release version="0.4" date="2020-09-09"></release>
+    <release version="0.3" date="2020-07-22"></release>
+    <release version="0.2" date="2020-05-13"></release>
+    <release version="0.1" date="2020-03-28"></release>
+  </releases>
+
+  <screenshots>
+    <screenshot type="default">
+      <image>https://gitlab.com/CalcProgrammer1/OpenRGB/-/wikis/uploads/e1d8d4603ecdd04f1acbcf6b2314fc66/OpenRGB_0.31_Device_View.PNG</image>
+    </screenshot>
+  </screenshots>
+
+  <keywords>
+    <keyword>rgb</keyword>
+    <keyword>utility</keyword>
+  </keywords>
+
+  <url type="bugtracker">https://gitlab.com/CalcProgrammer1/OpenRGB/-/issues</url>
+  <url type="contact">https://discord.gg/AQwjJPY</url>
+  <url type="donation">https://gitlab.com/CalcProgrammer1/OpenRGB/-/wikis/home#support-openrgb</url>
+  <url type="faq">https://gitlab.com/CalcProgrammer1/OpenRGB/-/wikis/Frequently-Asked-Questions</url>
+  <url type="help">https://gitlab.com/CalcProgrammer1/OpenRGB/-/wikis/home</url>
+  <url type="homepage">https://openrgb.org</url>
+</component>
-- 
GitLab


From 3d546908b7437c4ad18312c1decfef46d2fe34d6 Mon Sep 17 00:00:00 2001
From: Artem Polishchuk <ego.cordatus@gmail.com>
Date: Sun, 12 Dec 2021 19:46:22 +0300
Subject: [PATCH 2/3] OpenRGB.pro: Add AppStream metainfo installation

---
 OpenRGB.pro | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/OpenRGB.pro b/OpenRGB.pro
index 7dc2f2fd..04af0bba 100644
--- a/OpenRGB.pro
+++ b/OpenRGB.pro
@@ -1226,9 +1226,11 @@ unix:!macx {
     desktop.files+=qt/OpenRGB.desktop
     pixmap.path=$$PREFIX/share/pixmaps/
     pixmap.files+=qt/OpenRGB.png
+    metainfo.path=$$PREFIX/share/metainfo/
+    metainfo.files+=qt/org.openrgb.OpenRGB.metainfo.xml
     rules.path=$$PREFIX/lib/udev/rules.d/
     rules.files+=60-openrgb.rules
-    INSTALLS += target desktop pixmap rules
+    INSTALLS += target desktop pixmap metainfo rules
 }
 
 unix:!macx:CONFIG(asan) {
-- 
GitLab


From 2f0746a3019d498ea7cfd09bbef69ee04c17bca7 Mon Sep 17 00:00:00 2001
From: Artem Polishchuk <ego.cordatus@gmail.com>
Date: Sun, 12 Dec 2021 22:23:51 +0300
Subject: [PATCH 3/3] Update RPM .spec file due MR#934

---
 fedora/OpenRGB.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fedora/OpenRGB.spec b/fedora/OpenRGB.spec
index 98f5636b..0001af11 100644
--- a/fedora/OpenRGB.spec
+++ b/fedora/OpenRGB.spec
@@ -45,6 +45,7 @@ fi
 %{_datadir}/icons/hicolor/128x128/apps/%{_name}.png
 %{_datadir}/applications/%{_name}.desktop
 %{_datadir}/pixmaps/%{_name}.png
+%{_metainfodir}/org.%{name}.%{_name}.metainfo.xml
 /lib/udev/rules.d/60-%{name}.rules
 %license LICENSE
 %doc README.md
-- 
GitLab

